File: 242.valid-anagram.cpp
class Solution {
public:
bool isAnagram(string s, string t) {
    // Solution 1
    if(s.size()!=t.size()) return false;
    unordered_map<char,int> map_s;
    
    for(char i:s) 
        map_s[i]++;

    for(char j:t) 
        if(--map_s[j]<0) 
            return false;
    return true;
    
    // Solution 2
    // sort(s.begin(),s.end());
    // sort(t.begin(),t.end());
    // return s ==t;
    }
};

File: 242.valid-anagram.java
class Solution {
    // public boolean isAnagram(String s, String t) {
    // return sortString(s).equals(sortString(t));
    // }

    // public String sortString(String s){
    // char[] arr = s.toCharArray();
    // Arrays.sort(arr);
    // return new String(arr);
    // }

    public boolean isAnagram(String s, String t) {
        char[] freq = new char[26];

        for (char c : s.toCharArray()) {
            freq[c - 'a']++;
        }

        for (char c : t.toCharArray()) {
            freq[c - 'a']--;
        }

        for (int i : freq) {
            if (i != 0) {
                return false;
            }
        }

        return true;
    }
}